src/Storages/ObjectStorage/DataLakes/Iceberg/Mutations.cpp

```diff
@@ -611,48 +600,51 @@ void alter(
-            break;
-        case AlterCommand::Type::DROP_COLUMN:
-            if (params[0].clear)
-                throw Exception(ErrorCodes::BAD_ARGUMENTS, "Clear column is not supported for iceberg. Please use UPDATE instead");
-            metadata_json_generator.generateDropColumnMetadata(params[0].column_name);
-            break;
-        case AlterCommand::Type::MODIFY_COLUMN:
-            metadata_json_generator.generateModifyColumnMetadata(params[0].column_name, params[0].data_type);
-            break;
-        default:
-            throw Exception(ErrorCodes::LOGICAL_ERROR, "Unknown type of alter {}", params[0].type);
-    }
+        switch (params[0].type)
+        {
+            case AlterCommand::Type::ADD_COLUMN:
+                metadata_json_generator.generateAddColumnMetadata(params[0].column_name, params[0].data_type);
+                break;
+            case AlterCommand::Type::DROP_COLUMN:
+                if (params[0].clear)
+                    throw Exception(ErrorCodes::BAD_ARGUMENTS, "Clear column is not supported for iceberg. Please use UPDATE instead");
+                metadata_json_generator.generateDropColumnMetadata(params[0].column_name);
+                break;
+            case AlterCommand::Type::MODIFY_COLUMN:
+                metadata_json_generator.generateModifyColumnMetadata(params[0].column_name, params[0].data_type);
+                break;
+            default:
+                throw Exception(ErrorCodes::LOGICAL_ERROR, "Unknown type of alter {}", params[0].type);
+        }
 
-    std::ostringstream oss; // STYLE_CHECK_ALLOW_STD_STRING_STREAM
-    Poco::JSON::Stringifier::stringify(metadata, oss, 4);
-    std::string json_representation = removeEscapedSlashes(oss.str());
+        std::ostringstream oss; // STYLE_CHECK_ALLOW_STD_STRING_STREAM
+        Poco::JSON::Stringifier::stringify(metadata, oss, 4);
+        std::string json_representation = removeEscapedSlashes(oss.str());
 
-    auto [metadata_name, storage_metadata_name] = filename_generator.generateMetadataName();
-    auto cleanup = [] {};
-    Iceberg::writeMessageToFile(json_representation, storage_metadata_name, object_storage, context, cleanup);
+        auto [metadata_name, storage_metadata_name] = filename_generator.generateMetadataName();
 
-    if (configuration->getDataLakeSettings()[DataLakeStorageSetting::iceberg_use_version_hint].value)
-    {
-        auto filename_version_hint = filename_generator.generateVersionHint();
-        Iceberg::writeMessageToFile(storage_metadata_name, filename_version_hint.path_in_storage, object_storage, context, cleanup);
+        auto hint = filename_generator.generateVersionHint();
+        if (writeMetadataFileAndVersionHint(storage_metadata_name, json_representation, hint.path_in_storage, storage_metadata_name, object_storage, context, compression_method, configuration->getDataLakeSettings()[DataLakeStorageSetting::iceberg_use_version_hint]))
```

Comment:
As I remember we don't need to always write hint file - we have a setting `iceberg_use_version_hint`